### ISSUE TITLE
structured query validation error

### ISSUE THREAD

**Issue by sparnell** [joemfb/ml-common-ng#26]
There may be an issue in the structured query construction for range indexes. The return statement starting on line 901 in dist/ml-common-ng.js, 
```javascript
return {
      'range-constraint-query': {
        'constraint-name': name,
        'range-operator': operator || 'EQ',
        'value': asArray(values),
        'range-option': asArray(options)
      }
    };
  },
```
is causing a schema validation error due to the order of the properties (MarkLogic version 8.0-5.8). From what I can tell the order should be (constraint-name, value, range-operator, range-option). Changing the order fixed it for me. Can you confirm this?

Scott

**Comment by rjkennedy98** [comment]
Hi,
This is causing slush facets to fail. Is there a work around for this? Can we use a different version?
Richard

**Comment by grtjn** [comment]
Does it help to turn off options validation? It is one of the rest api properties.

**Comment by rjkennedy98** [comment]
@grtjn  Yes that is a valid work around. I do think this should be fixed anyways. One shouldn't have to turn off query validation for it to work.
